### PR TITLE
Update user profile fields

### PR DESCRIPTION
**Pull request by llewelld**
Updates the fields shown on the user profile page following the July 2023 meetup.

Contributes to issue #543.

### CODE REVIEW

**Comment by llewelld** [comment]
This should be in a reasonable state now, so I'm removing it from draft. I tried to cover all the elements described in [the issue](https://github.com/alan-turing-institute/AutSPACEs/issues/543#issuecomment-1625178816). The one part I've not tackled is the following:

> Additional questions: During the meeting we discussed whether to have a question on "(additional) relationship to autism", e.g. being a parent/carer/friend/co-worker etc. of someone with autism? I think generally it would be good to have this, maybe could fit in between Items 1 & 2?

I skipped this because I'd need a bit more direction on it: ideally the specific items to add to the drop down list, whether a free-text field is needed for any of the cases and whether there should be a 'make public' toggle. It looks like it also potentially relates to PR #554 that @obinnaofomah and @helendduncan are working on.

I'm very happy to make these changes, but if it makes it easier I could look at this in a separate PR.

File: server/settings/environments/development.py [hunk 1]
```diff
@@ -73,15 +73,35 @@ def _custom_show_toolbar(request):
...
+    'netdna.bootstrapcdn.com',
+    'cdn.jsdelivr.net',
+)
+CSP_STYLE_SRC = (
+    "'self'",
+    "'unsafe-inline'",
+    'fonts.googleapis.com',
+    'maxcdn.bootstrapcdn.com',
+    'netdna.bootstrapcdn.com',
+    'cdn.jsdelivr.net',
```
[the remark this answers is not in the document]
Comment (gedankenstuecke) [reply server/settings/environments/development.py:103]: I think that's fine, I guess we should at some point also check the production settings for that (hoping that this is a mental note enough to remember when things break during deployment 😂)

File: server/apps/main/templatetags/custom_tags.py [hunk 2]
```diff
@@ -32,3 +32,8 @@ def url_replace(request, field, value):
     query = request.GET.copy()
     query[field] = value
     return '?{}'.format(query.urlencode())
+
+@register.filter
+def public_toggle(items, index):
```
Comment (gedankenstuecke) [inline server/apps/main/templatetags/custom_tags.py:37]: I'm not 100% sure I understand what's going on here. What exactly does the public_toggle do/check for? :)

File: server/apps/users/forms.py [hunk 3]
```diff
@@ -12,47 +24,57 @@ class UserProfileForm(forms.Form):
...
-    autistic_identification.group = 1
 
-    description = forms.CharField(label="Description", max_length=2048, strip=True, required=False,
-                                 widget=forms.Textarea(attrs={"placeholder":"An optional description of yourself",
+    gender_self_identification = forms.CharField(label="Self identification",
+                                 max_length=150, strip=True, required=False,
+                                 widget=forms.TextInput())
+
+    gender_self_identification.group = 1
+    gender_self_identification.layout_horizontal = True
```
Comment (gedankenstuecke) [inline server/apps/users/forms.py:57]: The JS hide/show of the free-text field for self-identification works. But what doesn't seem to work (at least for me in Firefox) is actually filling in the text. The JS console doesn't give any error or anything, but I can't seem to actually type into the box 🙈

**Review by gedankenstuecke — COMMENTED** [review]
Thanks so much for this @llewelld, it looks really nice and I like the touch of the hover-info boxes for explaining why we collect that data. My only concern would be how well those work on mobile/how accessible they are, but we can test that later and simplify if needed! 

>>Additional questions: During the meeting we discussed whether to have a question on "(additional) relationship to autism", e.g. being a parent/carer/friend/co-worker etc. of someone with autism? I think generally it would be good to have this, maybe could fit in between Items 1 & 2?
>
>I skipped this because I'd need a bit more direction on it: ideally the specific items to add to the drop down list, whether a free-text field is needed for any of the cases and whether there should be a 'make public' toggle. It looks like it also potentially relates to PR https://github.com/alan-turing-institute/AutSPACEs/pull/554 that obinnaofomah and helendduncan are working on.

Yeah, I think it's fine to not incude these aspects in this inital PR, same for adding the display of public labels in the front-end of experiences! 

Otherwise, see inline comments for a couple of questions! 🙂

File: server/apps/users/forms.py (hunk 3 above)
Comment (llewelld) [reply server/apps/users/forms.py:57]: Ooops, thanks for checking. An error in refactoring meant the disabled flag got set in reverse. It should be fixed now.

File: server/apps/main/templatetags/custom_tags.py (hunk 2 above)
Comment (llewelld) [reply server/apps/main/templatetags/custom_tags.py]: This was used to condition on the 'Make public' fields. But it was hacky and I should have put more effort into coming up with a bettter alternative. I've made a more general filter for this now so all the toggle fields can be handled the same way on the form.

Comment (gedankenstuecke) [reply server/apps/main/templatetags/custom_tags.py]: Ah, so this is the bit which in this PR replaces the regular checkbox with the toggle switch, right? :)

**Review by gedankenstuecke — APPROVED** [review]
Looks and works great now! 👍

File: server/apps/main/templatetags/custom_tags.py (hunk 2 above)
Comment (llewelld) [reply server/apps/main/templatetags/custom_tags.py]: Yes; pretty much (the actual work is done in the template &mdash; which I know you know 😄 &mdash; but this helps simplify the conditioning).

**Pull request merged**